packages/@lwc/shared/src/signals.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isFalse } from './assert';
+
+let trustedSignals: WeakSet<object>;
+
+export function setSignalValidator(signals: WeakSet<object>) {
+    isFalse(trustedSignals, 'Already added a signal validator');
+
+    trustedSignals = signals;
+}
+
+export function addTrustedSignal(signal: object) {
+    trustedSignals?.add(signal);
+}
+
+export function isTrustedSignal(target: object): boolean {
+    return trustedSignals?.has(target);
```

Comment:
Actually, this implementation, along with the usage in the mutation tracker, would break OSS projects that don't call `setSignalValidator`. `isTrustedSignal` will always return false, and no signals will ever get subscribed to.

https://github.com/salesforce/lwc/blob/586318eb618e6eee06f526dbbccffd2dde148cb1/packages/%40lwc/engine-core/src/framework/mutation-tracker.ts#L45-L55

One possible solution is returning `boolean | undefined` from `isTrustedSignal` and checking `isTrustedSignal(target) !== false`.

Reply:
`isTrustedSignal` should return true if `setTrustedSignal` is never called. Also yes I agree that the return type should always be a boolean.

Reply:
I kinda skipped that in my head but agree it's good to be explicit, would have expected typescript to catch this.